File: 240410/왔다 갔던 구역 2/area-been-to-and-from2.py
N = int(input())

cur_pos = 0
visited = {}
for _ in range(N):
    size, mv_dir = input().split()
    size = int(size)
    if mv_dir == 'R':
        for i in range(size):
            if not cur_pos + i in visited.keys():
                visited[cur_pos + i] = 1
            else:
                visited[cur_pos + i] += 1
        cur_pos += size
    elif mv_dir == 'L':
        for i in range(1, size + 1):
            if not cur_pos - i in visited.keys():
                visited[cur_pos - i] = 1
            else:
                visited[cur_pos - i] += 1
        cur_pos -= size

cnts = 0
for pos, pos_cnts in visited.items():
    if pos_cnts >= 2:
        cnts += 1

print(cnts)

File: 260119/왔다 갔던 구역 2/area-been-to-and-from2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N;
    cin >> N;

    long long cur_pos = 0;
    unordered_map<long long, int> visited;
    visited.reserve(200000);

    for (int _ = 0; _ < N; _++) 
    {
        long long size;
        char mv_dir;
        cin >> size >> mv_dir;

        if (mv_dir == 'R') 
        {
            for (long long i = 0; i < size; i++) 
            {
                visited[cur_pos + i]++;  
            }
            cur_pos += size;
        } 
        else if (mv_dir == 'L') 
        {
            for (long long i = 1; i <= size; i++) 
            {
                visited[cur_pos - i]++;
            }
            cur_pos -= size;
        }
    }

    long long cnts = 0;
    for (auto &kv : visited) {
        if (kv.second >= 2) cnts++;
    }

    cout << cnts << "\n";
    return 0;
}